docs/dev/sql-nested-function-select-clause.md

```diff
@@ -0,0 +1,476 @@
+                                            "includes": [
+                                                "comment.data"
+                                            ],
+                                            "excludes": []
+                                        }
+                                    }
+                                }
+                            },
+                            {
+                                "nested": {
+                                    "query": {
+                                        "match_all": {
+                                            "boost": 1.0
+                                        }
+                                    },
+                                    "path": "message",
+                                    "ignore_unmapped": false,
+                                    "score_mode": "none",
+                                    "boost": 1.0,
+                                    "inner_hits": {
+                                        "ignore_unmapped": false,
+                                        "from": 0,
+                                        "size": 3,
+                                        "version": false,
+                                        "seq_no_primary_term": false,
+                                        "explain": false,
+                                        "track_scores": false,
+                                        "_source": {
+                                            "includes": [
+                                                "message.info"
+                                            ],
+                                            "excludes": []
+                                        }
+                                    }
+                                }
+                            }
+                        ],
+                        "adjust_pure_negative": true,
+                        "boost": 1.0
+                    }
+                }
+            ],
+            "adjust_pure_negative": true,
+            "boost": 1.0
+        }
+    },
+    "_source": {
+        "includes": [
+            "message.info",
+            "comment",
+            "comment.data",
+            "message"
+        ],
+        "excludes": []
+    }
+}
+```
+
+## 2 Architecture Diagrams
+### 2.1 Composite States for Nested Query Execution
+
+Nested function state diagram illustrating states in SQL plugin for push down execution. The nested operator stays in the `PhysicalPlan` after push down for flattening operation in post-processing. See section [2.3](#24-select-clause-nested-query-class-diagram) for flattening sequence and description.
+
+```mermaid
+stateDiagram-v2
+direction LR
+    LogicalPlan --> OptimizedLogicalPlan: Optimize
+    OptimizedLogicalPlan --> PhysicalPlan:push down
+    note right of PhysicalPlan
+      Note: NestedOperator stays in PhysicalPlan\nafter push down for post-processing.
```

Comment:
I'm not a big can of this `note`.  Would it be better to include a state for `Request` where the ExecutionEngine opens the request, and then we have another state afterwards for post-request processing?  It might be nice to include the inner_hits flattening there (as part of the `ProjectOperator`).

Reply:
No point in repeating things in multiple diagrams. Would like them to stay basic and only illustrate things that are necessary or unique for this query. I have removed the note and it is noted in the description that the `NestedOperator` stays in the `Physical Plan Tree` for post-processing, and a link to the sequence diagram that illustrates the response flattening.